logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/ObserveSelfUsersWithTeamsUseCase.kt

```diff
@@ -0,0 +1,35 @@
+package com.wire.kalium.logic.feature.user
+
+import com.wire.kalium.logic.data.session.SessionRepository
+import com.wire.kalium.logic.data.team.Team
+import com.wire.kalium.logic.data.user.SelfUser
+import com.wire.kalium.logic.feature.UserSessionScopeProvider
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.flatMapLatest
+
+interface ObserveSelfUsersWithTeamsUseCase {
+
+    suspend operator fun invoke(): Flow<List<Pair<SelfUser, Team?>>>
+}
+
+internal class ObserveSelfUsersWithTeamsUseCaseImpl internal constructor(
```

Comment:
the name is a bit confusing to me (ObserveSelfUsers -> return a list), maybe something like `ObserveVallidAccounts` is more clear 😅